src/auto/uri.rs

```diff
@@ -0,0 +1,530 @@
+        unsafe {
+            let mut error = ptr::null_mut();
+            let _ =
+                glib_sys::g_uri_is_valid(uri_string.to_glib_none().0, flags.to_glib(), &mut error);
+            if error.is_null() {
+                Ok(())
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn join(
+        flags: UriFlags,
+        scheme: Option<&str>,
+        userinfo: Option<&str>,
+        host: Option<&str>,
+        port: i32,
+        path: &str,
+        query: Option<&str>,
+        fragment: Option<&str>,
+    ) -> Option<GString> {
+        unsafe {
+            from_glib_full(glib_sys::g_uri_join(
+                flags.to_glib(),
+                scheme.to_glib_none().0,
+                userinfo.to_glib_none().0,
+                host.to_glib_none().0,
+                port,
+                path.to_glib_none().0,
+                query.to_glib_none().0,
+                fragment.to_glib_none().0,
+            ))
+        }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn join_with_user(
+        flags: UriFlags,
+        scheme: Option<&str>,
+        user: Option<&str>,
+        password: Option<&str>,
+        auth_params: Option<&str>,
+        host: Option<&str>,
+        port: i32,
+        path: &str,
+        query: Option<&str>,
+        fragment: Option<&str>,
+    ) -> Option<GString> {
+        unsafe {
+            from_glib_full(glib_sys::g_uri_join_with_user(
+                flags.to_glib(),
+                scheme.to_glib_none().0,
+                user.to_glib_none().0,
+                password.to_glib_none().0,
+                auth_params.to_glib_none().0,
+                host.to_glib_none().0,
+                port,
+                path.to_glib_none().0,
+                query.to_glib_none().0,
+                fragment.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn list_extract_uris(uri_list: &str) -> Vec<GString> {
+        unsafe {
+            FromGlibPtrContainer::from_glib_full(glib_sys::g_uri_list_extract_uris(
+                uri_list.to_glib_none().0,
+            ))
+        }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn parse(uri_string: &str, flags: UriFlags) -> Result<Uri, Error> {
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret =
+                glib_sys::g_uri_parse(uri_string.to_glib_none().0, flags.to_glib(), &mut error);
+            if error.is_null() {
+                Ok(from_glib_full(ret))
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
+
+    //#[cfg(any(feature = "v2_66", feature = "dox"))]
+    //pub fn parse_params(params: &str, separators: &str, flags: UriParamsFlags) -> Result</*Unknown conversion*//*Unimplemented*/HashTable TypeId { ns_id: 0, id: 28 }/TypeId { ns_id: 0, id: 28 }, Error> {
```

Comment:
That's strange. We support specifically `HashTable<String, String>` (and nothing else)